libcst/codemod/visitors/_gather_exports.py

```diff
@@ -43,68 +43,87 @@ def __init__(self, context: CodemodContext) -> None:
+                    if isinstance(element_value, (cst.List, cst.Tuple, cst.Set)):
+                        self._is_assigned_export.add(value)
+                        self._is_assigned_export.add(element_value)
+                        return True
         return False
 
-    def leave_Assign(self, original_node: libcst.Assign) -> None:
-        self._in_assignment -= 1
-
-    def visit_List(self, node: libcst.List) -> bool:
-        self._in_list += 1
-        # Only visit list/set entries when we're in an __all__
-        # assignment. We gate also by internal counters, so this
-        # is simply an optimization.
-        return self._in_assignment == 1 and self._in_list == 1
-
-    def leave_List(self, original_node: libcst.List) -> None:
-        self._in_list -= 1
-
-    def visit_Tuple(self, node: libcst.Tuple) -> bool:
-        self._in_list += 1
-        # Only visit list/set entries when we're in an __all__
-        # assignment. We gate also by internal counters, so this
-        # is simply an optimization.
-        return self._in_assignment == 1 and self._in_list == 1
-
-    def leave_Tuple(self, original_node: libcst.Tuple) -> None:
-        self._in_list -= 1
-
-    def visit_Set(self, node: libcst.Set) -> bool:
-        # Only visit list/set entries when we're in an __all__
-        # assignment. We gate also by internal counters, so this
-        # is simply an optimization.
-        self._in_list += 1
-        return self._in_assignment == 1 and self._in_list == 1
-
-    def leave_Set(self, original_node: libcst.Set) -> None:
-        self._in_list -= 1
-
-    def visit_Element(self, node: libcst.Element) -> bool:
-        # See if this is a entry that is a string.
-        extraction = self.extract(
-            node, m.Element(m.SaveMatchedNode(m.SimpleString(), "string"))
-        )
-        if extraction:
-            string = ensure_type(extraction["string"], libcst.SimpleString)
-            self.explicit_exported_objects.add(string.evaluated_value)
-
-        # Don't need to visit children
+    def visit_List(self, node: cst.List) -> bool:
+        if node in self._is_assigned_export:
+            self._in_assigned_export.add(node)
+            return True
+        return False
+
+    def leave_List(self, original_node: cst.List) -> None:
+        self._is_assigned_export.discard(original_node)
+        self._in_assigned_export.discard(original_node)
+
+    def visit_Tuple(self, node: cst.Tuple) -> bool:
+        if node in self._is_assigned_export:
+            self._in_assigned_export.add(node)
+            return True
         return False
+
+    def leave_Tuple(self, original_node: cst.Tuple) -> None:
+        self._is_assigned_export.discard(original_node)
+        self._in_assigned_export.discard(original_node)
+
+    def visit_Set(self, node: cst.Set) -> bool:
+        if node in self._is_assigned_export:
+            self._in_assigned_export.add(node)
+            return True
+        return False
```

Comment:
This line is not covered by tests.
It can be easily covered by having a test case like

```
__all__, _ = ["bar", "baz"], ["biz"], {"foo"}
```